src/Metadata/Resource/Factory/ParameterResourceMetadataCollectionFactory.php

```diff
@@ -121,6 +142,74 @@ private function setDefaults(string $key, Parameter $parameter, string $resource
             }
         }
 
+        $schema = $parameter->getSchema() ?? $parameter->getOpenApi()?->getSchema();
+        if (!$parameter->getConstraints()) {
+            $parameter = $this->addSchemaValidation($parameter, $schema, $parameter->getRequired() ?? $description['required'] ?? false, $parameter->getOpenApi());
+        }
+
         return $parameter;
     }
+
+    private function addSchemaValidation(Parameter $parameter, ?array $schema = null, bool $required = false, ?OpenApiParameter $openApi = null): Parameter
```

Comment:
Couldn't we make the dependency to `symfony/validator` optional, or even better, use inversion of control to move this method in the Symfony bridge (and add an extension point to allow the user to use another validation library)?

Reply:
Ok, this is premature as we require the Symfony `Constraint` class anyway. Let's keep this as-is for now.